### DISCUSSION TITLE
Discussion on the correct behavior of the batch acquisition method

### DISCUSSION THREAD

**Discussion by matthewcarbone** [ziatdinovmax/gpax#57] [Q&A]
@ziatdinovmax In BoTorch, the batch acquisition methods allow one to set a value for `q` such that the best combination of `q` points that _jointly_ optimize your acquisition function are chosen. It does not appear that GPax is doing the same thing. It seems that a subsample of the provided points are being fed to the single-`q` acquisition function independently. So this is more just a parallel batch evaluation of the acquisition function, it's not a joint optimization.

It might be good to have this feature. It would probably require some work though, and it will almost certainly require an option to pass `X=None` on acquisition functions, pass bounds instead, and call some sort of MC sampler to sample combinations of points in the space.

What do you think about this? Clearly the most common use cases for GPax are sequential experimentation so this might be an edge case for the user base.

**Comment by ziatdinovmax** [comment]
We can implement that as well. It shouldn't be difficult. Let's add it to the v0.3 milestone.
